### PR TITLE
Tidy kotest-common logger

### PR DESCRIPTION
**Pull request by aSemy**
- Annotate vals/funs with `@KotestInternal` to avoid opt-in warnings
- Remove `@OverloadResolutionByLambdaReturnType` from log function (it has no overloads)

I saw that the failure in [this build](https://github.com/kotest/kotest/actions/runs/9431572967/job/25980529942?pr=4076) had an error relating to the logger functions, so maybe tidying it up will help.

```stacktrace
org.gradle.api.internal.tasks.testing.TestSuiteExecutionException: Could not complete execution for Gradle Test Executor 1.
	[...]
Caused by: org.junit.platform.commons.JUnitException: TestEngine with ID 'kotest' failed to discover tests
	[...]
Caused by: java.lang.NoSuchMethodError: 'void io.kotest.mpp.Logger.logsimple(kotlin.jvm.functions.Function0)'
	at io.kotest.runner.junit.platform.KotestJunitPlatformTestEngine.discover(KotestJunitPlatformTestEngine.kt:101)
	at io.kotest.runner.junit.platform.KotestJunitPlatformTestEngine.discover(KotestJunitPlatformTestEngine.kt:35)
	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverEngineRoot(EngineDiscoveryOrchestrator.java:152)
	... 28 more
```

### CODE REVIEW

File: kotest-common/src/commonMain/kotlin/io/kotest/common/annotations.kt [hunk 1]
```diff
@@ -16,7 +16,12 @@ annotation class DelicateKotest
 /**
  * An internal Kotest feature that is public for operational reasons but should not be used by end users.
  */
-@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.TYPEALIAS)
+@Target(
+   AnnotationTarget.CLASS,
+   AnnotationTarget.FUNCTION,
+   AnnotationTarget.TYPEALIAS,
+   AnnotationTarget.PROPERTY,
```
Comment (Kantis) [inline kotest-common/src/commonMain/kotlin/io/kotest/common/annotations.kt:23]: You didn't actually add the annotation on a property in the end? Should we revert the change here?

Comment (aSemy) [reply kotest-common/src/commonMain/kotlin/io/kotest/common/annotations.kt:23]: `AnnotationTarget.PROPERTY` is required for adding `@KotestInternal` to `val start` in `logger.kt`

Comment (Kantis) [reply kotest-common/src/commonMain/kotlin/io/kotest/common/annotations.kt:23]: Oh, right :) carry on

**Review by Kantis — APPROVED** [review]
